src/Components/BehaviorsModal/Internal/Widgets/DataHistoryWidget/DataHistoryWidget.tsx

```diff
@@ -106,11 +108,13 @@ const DataHistoryWidget: React.FC<IDataHistoryWidgetProps> = ({
     const xMaxDateInMillisRef = useRef<number>(null);
 
     const updateXMinAndMax = useCallback(() => {
-        const nowInMillis = Date.now();
+        const nowInMillis = isChromatic()
```

Comment:
Since this is coming from a dev/test tool, I think it should be passed in via `props` or a `useContext`

Reply:
Added context and called isChromatic at the very top most level of Storybook in preview.js checking if it is Chromatic environment wrapping the story with stable date. And using a hook (which internally calls useContext) in the components I am checking if that stable date exist, if exist use it, if not use Date.now as the end date to show in the Highcharts.